master/internal/core.go

```diff
@@ -791,6 +793,67 @@ func (m *Master) restoreNonTerminalExperiments() error {
 	return nil
 }
 
+func (m *Master) restoreGenericTasks(ctx context.Context) error {
+	snapshots := []command.CommandSnapshot{}
+	err := db.Bun().NewSelect().Model(&snapshots).
+		Relation("Allocation").
+		Relation("Task").
+		Relation("Task.Job").
+		Where("allocation.end_time IS NULL").
+		Where("allocation.state != ?", model.AllocationStateTerminated).
+		Where("task.task_id = command_snapshot.task_id").
+		Where("task.task_type = ?", model.TaskTypeGeneric).
+		Where("command_snapshot.generic_task_spec IS NOT NULL").
+		Scan(ctx)
+	if err != nil {
+		return err
+	}
+
+	for i := range snapshots {
+		taskID := snapshots[i].TaskID
+		jobID := snapshots[i].Task.JobID
+
+		logCtx := logger.Context{
+			"job-id":    jobID,
+			"task-id":   taskID,
+			"task-type": snapshots[i].Task.TaskType,
+		}
+
+		priorityChange := func(priority int) error {
+			return nil
+		}
+		if err := tasklist.GroupPriorityChangeRegistry.Add(*jobID, priorityChange); err != nil {
+			return err
+		}
+
+		onAllocationExit := m.getGenericTaskOnAllocationExit(ctx, taskID, *jobID, logCtx)
+
+		isSingleNode := snapshots[i].GenericTaskSpec.GenericTaskConfig.Resources.IsSingleNode() != nil &&
+			*snapshots[i].GenericTaskSpec.GenericTaskConfig.Resources.IsSingleNode()
```

Comment:
Can `IsSingleNode` handle this?

Reply:
We want `IsSingleNode` to default to false if it is not provided. Essentially is it's a null pointer then assume false